pkg/services/store/entity/sqlstash/sql_storage_server.go

```diff
@@ -1021,15 +1021,21 @@ func (s *sqlEntityServer) List(ctx context.Context, r *entity.EntityListRequest)
 				return nil, err
 			}
 
-			args = append(args, key.Namespace, key.Group, key.Resource)
-			whereclause := "(" + s.dialect.Quote("namespace") + "=? AND " + s.dialect.Quote("group") + "=? AND " + s.dialect.Quote("resource") + "=?"
-			if key.Name != "" {
-				args = append(args, key.Name)
-				whereclause += " AND " + s.dialect.Quote("name") + "=?"
-			}
-			whereclause += ")"
+			if key.Namespace != "" {
+				args = append(args, key.Namespace, key.Group, key.Resource)
+				whereclause := "(" + s.dialect.Quote("namespace") + "=? AND " + s.dialect.Quote("group") + "=? AND " + s.dialect.Quote("resource") + "=?"
+				if key.Name != "" {
```

Comment:
instead of duplicating the whole block, let's just treat namespace the same way as name